### PR TITLE
Lock down the Instance Meta-Data Service (IMDS)

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description ##

This pull request:
* Explicitly enables IMDS (which is the default behavior)
* Requires the use of IMDSv2
* Restricts the retrieval of IMDSv2 tokens to the host (also the default behavior)

## 💭 Motivation and context ##

See cisagov/cool-system#174 and cisagov/cool-system#175.

## 🧪 Testing ##

All `pre-commit` hooks pass.

I built [a new Guacamole AMI](https://github.com/cisagov/guacamole-packer/pull/43) containing the changes from cisagov/ansible-role-freeipa-client#28 and was able to verify that it functioned as expected.  Later I noticed that [the OpenVPN AMI with similar changes](https://github.com/cisagov/openvpn-packer/pull/44) _did not_ function as expected, and I realized that the Guacamole AMI was only behaving because of cisagov/guacamole-packer#44.

I am currently to get a newer version of `python3-botocore` in Debian Stable or Stable Backports.  In the meantime the Guacamole AMI is working and I have a workaround for the OpenVPN AMI.  That workaround is to:
1. Deploy the OpenVPN instance
2. Manually [make the use of IMDSv2 optional via the AWS CLI](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html#configuring-IMDS-existing-instances)
3. Manually re-run the cloud-init scripts for the OpenVPN instance

Therefore I think it is safe and better to proceed with merging this PR, rather than to leave it dangling while I continue to work through the Debian Backports process. 

## ✅ Checklist ##

* [x] This PR has an informative and human-readable title.
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] All relevant type-of-change labels have been added.
* [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
* [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
* [x] All new and existing tests pass.

### CODE REVIEW

**Review by dav3r — APPROVED** [review]
🔒 Approved, though I would feel better if we did a staging deployment of some of these instances like Guac and Kali to make sure that they don't have any unexpected issues.

**Comment by jsf9k** [comment]
>  Approved, though I would feel better if we did a staging deployment of some of these instances like Guac and Kali to make sure that they don't have any unexpected issues.

I plan to do that, once these PRs are merged; new AMIs are built for FreeIPA, OpenVPN, and Guacamole; and the merged code and new AMIs are deployed to COOL staging:
* cisagov/ansible-role-freeipa-server#37
* cisagov/ansible-role-freeipa-client#28
* cisagov/freeipa-server-tf-module#46
* cisagov/openvpn-server-tf-module#60

I want to verify that the one hop jazz doesn't goof up anything inside of a Docker container that is trying to talk to the IMDS.

**Review by mcdonnnj — APPROVED** [review]
🔒🔒🔒 Approved with agreement on testing for verification.

**Comment by jsf9k** [comment]
@dav3r and @mcdonnnj, please let me know if you are OK with going ahead and merging this after reading the new content in the "Testing" section of the PR description.

**Comment by dav3r** [comment]
> @dav3r and @mcdonnnj, please let me know if you are OK with going ahead and merging this after reading the new content in the "Testing" section of the PR description.

@jsf9k - I am good with this approach, thanks! 👍

**Comment by mcdonnnj** [comment]
> 
> 
> @dav3r and @mcdonnnj, please let me know if you are OK with going ahead and merging this after reading the new content in the "Testing" section of the PR description.

@jsf9k This seems like a reasonable approach to me. I'm on board 👍

**Pull request merged**
